tools/flight_data_downloader/fdd.py

```diff
@@ -0,0 +1,113 @@
+import signal
+
+from fdd_transport import FDDTransport
+from serial_transport import SerialTransport, print_serial_help
+from tftp_transport import TFTPTransport, print_tftp_help
+from print_colors import print_red, print_green
+
+transport = FDDTransport()
+
+def handle_set_command(args):
+    global transport
+
+    attribute = args[0]
+    args = args[1:]
+
+    if attribute == "transport":
+        if args[0] == "tftp":
+            transport = TFTPTransport()
+            print_green("Set to TFTP.")
+        elif args[0] == "serial":
+            transport = SerialTransport()
+            print_green("Set to serial.")
+        else:
+            print_red("Invalid transport")
+        return
+
+    if attribute == "output":
+        if len(args) == 1:
+            transport.set_output_folder(args[0])
+            print_green("Set output folder to {}".format(args[0]))
+        elif len(args) == 0:  # No folder specified
+            print_red("Output folder not specified")
+        else:  # Spaces in folder name
+            print_red("Invalid output folder")
+
+        return
+
+    if transport is not None and not isinstance(transport, FDDTransport):
```

Comment:
this returns false for serial transport (and presumably tfttransport) bc isinstance checks both type and subclasses. I think you just want `is` here to make sure its not the abstract. (or maybe check if it isinstance of FDDTransport and isnt FDDTransport itself